### DISCUSSION TITLE
One view VM not able to create server profile

### DISCUSSION THREAD

**Discussion by prabhum456** [HewlettPackard/POSH-HPOneView#546] [Q&A] [answered]
On One view VM not able to create server profile 

New-HPOVServerProfile -name Management_Server_sms03_Server_Profile -ServerProfileTemplate @{type=ServerProfileTemplateV7; uri=/rest/server-profile-templates/29e209e1-5609-41f9-aa7e-afd6cee10765; name=Cx750_DL360_Gen10_Management_Server_Template; description=Cx750 DL360 Gen10 Management Server Template; serverProfileDescription=Cx750 DL360 Gen10 Management Server Profile; serverHardwareTypeUri=/rest/server-hardware-types/F67FE1C2-D665-4C30-9221-38F2A88CDD64; enclosureGroupUri=; affinity=; hideUnusedFlexNics=; macType=Physical; wwnType=Physical; serialNumberType=Physical; iscsiInitiatorNameType=AutoGenerated; osDeploymentSettings=; firmware=; connectionSettings=; bootMode=; boot=; bios=; managementProcessor=; localStorage=; sanStorage=; category=server-profile-templates; created=2021-01-19T13:36:18.324Z; modified=2021-01-19T13:36:18.335Z; status=OK; state=; scopesUri=/rest/scopes/resources/rest/server-profile-templates/29e209e1-5609-41f9-aa7e-afd6cee10765; eTag=1611063378335/1; ApplianceConnection=x.x.x.x} -server  -description (@{type=ServerProfileTemplateV7; uri=/rest/server-profile-templates/29e209e1-5609-41f9-aa7e-afd6cee10765; name=Cx750_DL360_Gen10_Management_Server_Template; description=Cx750 DL360 Gen10 Management Server Template; serverProfileDescription=Cx750 DL360 Gen10 Management Server Profile; serverHardwareTypeUri=/rest/server-hardware-types/F67FE1C2-D665-4C30-9221-38F2A88CDD64; enclosureGroupUri=; affinity=; hideUnusedFlexNics=; macType=Physical; wwnType=Physical; serialNumberType=Physical; iscsiInitiatorNameType=AutoGenerated; osDeploymentSettings=; firmware=; connectionSettings=; bootMode=; boot=; bios=; managementProcessor=; localStorage=; sanStorage=; category=server-profile-templates; created=2021-01-19T13:36:18.324Z; modified=2021-01-19T13:36:18.335Z; status=OK; state=; scopesUri=/rest/scopes/resources/rest/server-profile-templates/29e209e1-5609-41f9-aa7e-afd6cee10765; eTag=1611063378335/1; ApplianceConnection=x.x.x.x}.ServerProfileDescription) -Async -confirm:False -force:True
Server Profile [Management_Server_sms03_Server_Profile] creation failed with error:Cannot validate argument on parameter 'Server'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again..

Creation of server profile for DL servers Failed
Management_Server_sms03_Server_Profile	Fail	Creation of server profile for DL servers Failed
**1/19/2021 8:21:27 PM	Move-VM		Exception has been thrown by the target of an invocation.**

**Comment by DungKHoang** [comment, marked as answer]
When you create a server profile, there must be a server hardware specified in the parameter -server.
According to the error message, this parameter is NOT specified, thus the failure
"Cannot validate argument on parameter 'Server'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again"

**Reply by prabhum456** [reply]
Thanks, @DungKHoang let me check

**Reply by ChrisLynchHPE** [reply]
If you are specyfing `-AssignmentType` to `Server` when creating a server profile, you must specify a server resource to assign to using the `-Server` parameter.  This is documented in the Cmdlet help, [New-OVServerProfile](https://hpe-docs.gitbook.io/posh-hpeoneview/cmdlets/v5.50/servers/new-ovserverprofile) and [-AssignmentType](https://hpe-docs.gitbook.io/posh-hpeoneview/cmdlets/v5.50/servers/new-ovserverprofile#assignmenttype-less-than-string-greater-than) parameter.

**Reply by prabhum456** [reply]
Thanks @ChrisLynchHPE , can close this. 
the server name was blank on ilo
